vars/cloudFoundryDeploy.groovy

```diff
@@ -0,0 +1,166 @@
+import com.sap.piper.Utils
+import com.sap.piper.ConfigurationHelper
+
+import groovy.transform.Field
+
+@Field String STEP_NAME = 'cloudFoundryDeploy'
+@Field Set STEP_CONFIG_KEYS = [
+    'cloudFoundry',
+    'deployUser',
+    'deployTool',
+    'deployType',
+    'dockerImage',
+    'dockerWorkspace',
+    'mtaDeployParameters',
+    'mtaExtensionDescriptor',
+    'mtaPath',
+    'smokeTestScript',
+    'smokeTestStatusCode',
+    'stashContent']
+@Field Map CONFIG_KEY_COMPATIBILITY = [cloudFoundry: [apiEndpoint: 'cfApiEndpoint', appName:'cfAppName', credentialsId: 'cfCredentialsId', manifest: 'cfManifest', org: 'cfOrg', space: 'cfSpace']]
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+def call(Map parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+
+        def utils = parameters.juStabUtils
+        if (utils == null) {
+            utils = new Utils()
+        }
+
+        def script = parameters.script
+        if (script == null)
+            script = [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        Map config = ConfigurationHelper
+            .loadStepDefaults(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS, this, CONFIG_KEY_COMPATIBILITY)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS, this, CONFIG_KEY_COMPATIBILITY)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS, this, CONFIG_KEY_COMPATIBILITY)
+            .mixin(parameters, PARAMETER_KEYS, this, CONFIG_KEY_COMPATIBILITY)
+            .dependingOn('deployTool').mixin('dockerImage')
+            .dependingOn('deployTool').mixin('dockerWorkspace')
+            .withMandatoryProperty('cloudFoundry/org')
+            .withMandatoryProperty('cloudFoundry/space')
+            .use()
+
+        echo "[${STEP_NAME}] General parameters: deployTool=${config.deployTool}, deployType=${config.deployType}, cfApiEndpoint=${config.cloudFoundry.apiEndpoint}, cfOrg=${config.cloudFoundry.org}, cfSpace=${config.cloudFoundry.space}, cfCredentialsId=${config.cloudFoundry.credentialsId}, deployUser=${config.deployUser}"
+
+        utils.unstash 'deployDescriptor'
+
+        if (config.deployTool == 'mtaDeployPlugin') {
+            // set default mtar path
+            config = new ConfigurationHelper(config)
+                .addIfEmpty('mtaPath', config.mtaPath?:findMtar())
+                .use()
+
+            dockerExecute(dockerImage: config.dockerImage, dockerWorkspace: config.dockerWorkspace, stashContent: config.stashContent) {
+                deployMta(config)
+            }
+            return
+        }
+
+        if (config.deployTool == 'cf_native') {
+            def smokeTest = ''
```

Comment:
There seems to be no read access to variable ```smokeTest```.

Reply:
well spotted. This needs to go into the config ...